ricecooker/utils/SCORM_metadata.py

```diff
@@ -0,0 +1,173 @@
+            )
+
+        if le_utils_type and le_utils_type not in le_utils_activities:
+            le_utils_activities.append(le_utils_type)
+
+    return le_utils_activities
+
+
+# Define mappings from SCORM educational types to educator-focused resource types
+SCORM_to_resource_type_mappings = {
+    "exercise": resource_type.EXERCISE,
+    "simulation": resource_type.ACTIVITY,
+    "questionnaire": resource_type.ACTIVITY,
+    "diagram": resource_type.MEDIA,
+    "figure": resource_type.MEDIA,
+    "graph": resource_type.MEDIA,
+    "index": resource_type.GUIDE,
+    "slide": resource_type.LESSON,
+    "table": resource_type.TUTORIAL,
+    "narrative text": resource_type.TEXTBOOK,
+    "exam": resource_type.EXERCISE,
+    "experiment": resource_type.ACTIVITY,
+    "problem statement": resource_type.ACTIVITY,
+    "self assessment": resource_type.ACTIVITY,
+    "lecture": resource_type.LESSON,
+}
+
+
+# Mapping for intendedEndUserRole when the resource is for educators
+SCORM_intended_role_to_resource_type_mapping = {
+    "teacher": resource_type.LESSON_PLAN,
+    "author": resource_type.GUIDE,
+    "manager": resource_type.GUIDE,
+}
+
+
+def map_scorm_to_educator_resource_types(metadata_dict):
+    educator_resource_types = []
+
+    # Extract the learning resource types and intended end user role from the SCORM data
+    learning_resource_types = metadata_dict.get("learningResourceType", [])
+    intended_roles = metadata_dict.get("intendedEndUserRole", [])
+
+    # Map each SCORM type to an educator-focused resource type
+    for learning_resource_type in learning_resource_types:
+        mapped_type = SCORM_to_resource_type_mappings.get(learning_resource_type)
+        if mapped_type and mapped_type not in educator_resource_types:
+            educator_resource_types.append(mapped_type)
+
+    # Check if the intended end user role matches any educator roles
+    for role in intended_roles:
+        if (
+            role in SCORM_intended_role_to_resource_type_mapping
+            and SCORM_intended_role_to_resource_type_mapping[role]
+            not in educator_resource_types
+        ):
+            educator_resource_types.append(
+                SCORM_intended_role_to_resource_type_mapping[role]
+            )
+
+    return educator_resource_types
+
+
+def infer_beginner_level_from_difficulty(metadata_dict):
+    # Beginner difficulty levels
+    beginner_difficulties = {"very easy", "easy"}
+
+    # Check if the difficulty level indicates beginner content
+    difficulty = metadata_dict.get("difficulty")
+    if difficulty in beginner_difficulties:
+        return [needs.FOR_BEGINNERS]
+
+    return []
+
+
+def update_node_from_metadata(node, metadata_dict):
+    # Update the node with the general metadata
+    node.description = metadata_dict.get("description") or node.description
+    if metadata_dict.get("language"):
+        node.set_language(metadata_dict.get("language"))
```

Comment:
I had two problems ithe these few lines of code:
language was en-US
metadata.dict was this one:
```
(Pdb) metadata_dict
{'title': 'Work with computers', 'description': 'This learning path will introduce you to the different parts and types of the computer and their functions. You will also learn the difference between operating systems and applications and their functions. Peripherals and portable storage devices will be discussed as well.', 'language': 'en-US', 'keyword': 'Default'}
```

It worked with these changes (I ignore if the `keyword` change will work with other scorm files)
```
    if metadata_dict.get("language"):
        node.set_language(metadata_dict.get("language").lower().split("-")[0])
    keyword = metadata_dict.get("keyword")
    if keyword:
        node.tags = node.tags + [keyword,]
```

Reply:
Updated - hopefully has a similar effect, add some additional validation to make sure it's a language we know.